js/server/modules/@arangodb/foxx/test-utils.js

```diff
@@ -31,9 +31,11 @@ exports.createNativeRequest = function (opts) {
     suffix: opts.path ? opts.path.split('/').filter(Boolean) : [],
     headers: opts.headers || {},
     cookies: opts.cookies || {},
+    portType: opts.socketPath ? 'unix' : (opts.protocol || 'http'),
```

Comment:
FYI this is just for the test dummies. The dummy data is actually a bit inconsistent (arangod sets the port to `0` when in socket mode) but it's sufficient to trigger the behavior.